### PR TITLE
fix:  lambdas and parens, multiple indents

### LINKED ISSUE #346: lambdas as expressions are incomplete
This may be known based on a commit 3 months ago, having a look myself:
# WORKS
```
  var pure_test := func():
    if 1:
      var test := 1
      pass
```
# WORKS
```
  stack(func():
    if 1:
      var test := 1)
```
# DOES NOT WORK
```
  stack(func():
    if 1:
      var test := 1
      pass)
```
(or any other 2nd statement)

### PR DESCRIPTION
**Pull request by ludamad**
Closes #346

Background: Thanks for making this tool. Lambda bugs were getting in the way in some cases, and I'd like to adopt it.
It seems INDENT and DEDENT were not emitted inside a lambda, other than one level deep.
I dove in and tried to fix it. The big test suite was a help as I thought I was done a few times.

I still noticed edge-cases with lambda, but mostly stuff like

```
var a = {
   f = func():  for x in y: print("hey")
}
```
which I looked at and had some ideas, like generating a virtual newline, but am leaving it here as I can work the codebase into shape now within the edge-cases

### CODE REVIEW

File: gdtoolkit/parser/gdscript.lark [hunk 1]
```diff
@@ -255,18 +255,8 @@ _lambda_suite: _lambda_body
              | _standalone_lambda_stmt
 _lambda_body: _NL _INDENT _func_stmt+ _DEDENT
 _standalone_lambda_stmt: _simple_func_stmt
-                       // | annotation* compound_lambda_stmt
```
Comment (ludamad) [inline gdtoolkit/parser/gdscript.lark:258]: opinionated, but I think fixing compound func stmt is the way to go and this works *some* of the time. could def revert this part tho

**Comment by ogrady** [comment]
@Scony can you please take a look at this proposal? We too are running into the issue this PR is addressing and would love to see it merged. Thanks!

**Comment by ludamad** [comment]
I welcome you in the meanwhile to try installing this specific commit `pip3 install git+https://github.com/Scony/godot-gdscript-toolkit.git@57b1693a28ff49f4ec45f871016831050a800ee5` - I will be happy to give support if things go wrong :)

**Comment by Scony** [comment]
> @Scony can you please take a look at this proposal? We too are running into the issue this PR is addressing and would love to see it merged. Thanks!

Got this in my queue, but it will take some time.

**Comment by Scony** [comment]
Thanks!
